### ISSUE TITLE
Chart data does not show up when changing type with series.update() from "line" type to "ohlc" or "candlestick"

### ISSUE THREAD

**Issue by svensigmond** [highcharts/highcharts#11219]
#### Expected behaviour
When updating a series from the "_line_" type to "_ohlc_" or "_candlestick_" through the use of `series.update()`, the chart and data should be redrawn and the data should be correctly displayed.

#### Actual behaviour
When updating a series from the "_line_" type to "_ohlc_" or "_candlestick_" through the use of `series.update()`, the chart is updated, but the data is not visible in the chart. Adjusting the zoom level through the use of the navigator sometimes displays the data for smaller zoom ranges.

#### Live demo with steps to reproduce
Please use the selectbox at the bottom left of the chart to change the type from line to ohlc or candlestick.

[Broken in 7.1.2](https://codepen.io/ssigmond/pen/EBoNKr)

[Works in 6.2.0](https://codepen.io/ssigmond/pen/agEpMw)

#### Product version
Highstock 7.1.2

#### Affected browser(s)
- Firefox 67 
- Chrome 75
- Probably other browsers too

**Comment by sebastianbochan** [comment]
Hi @svensigmond,
Thank you for reporting about the problem. Its a regression since v7.0.3 version.

**Internal note**
Bisect leads me to https://github.com/highcharts/highcharts/commit/8e6a65e

**Comment by pawelfus** [comment]
**Workaround:**
Add `dataGrouping.approximation` information: 
```js
    series.update({
      type,
      dataGrouping: {
        approximation: type === 'line' ? 'average' : 'ohlc'
      }
    }, true);
```

**Comment by TorsteinHonsi** [comment]
I think this is an issue we need to resolve on a general level. First of all, we need to continue supporting `series.update({ type })`, removing it will be a big pain for many users.

I think, out of the proposed solutions above, two are worth following up:
* Using `setPrototypeOf`. The main downside listed on MDN is performance, but I think in our case that shouldn't matter much, as we are typically casting one or a few series in the chart. I think we should test this option.
* If that doesn't work, implement our own type check. I have been thinking out this as a shorthand to `instanceof`, for example `series.is('column')`. But since `instanceof` doesn't survive casting, we could add our own inheritance chain, so each new series type (in the `seriesType` factory), adds a trace of its ancestors. So for example the column range will have a property like `this.inheritanceChain = ['line', 'area', 'arearange', 'columnrange']`, and the `is()` function would just do and `indexOf` in that chain.

**Comment by khlieng** [comment]
Fixed in #14744.
